### PR TITLE
TcpDuplexConnection supports multiple subscribes

### PR DESCRIPTION
**Pull request by lehecka**

### CODE REVIEW

**Comment by lehecka** [comment]
All implementations of DuplexConnection should support repeated connecting to input/output.
This is doing it for the TcpDuplexConnection.

File: src/tcp/TcpDuplexConnection.cpp [hunk 1]
```diff
@@ -122,36 +116,113 @@ class TcpReaderWriter : public ::folly::AsyncTransportWrapper::WriteCallback,
...
+  TcpOutputSubscriber(
+      std::shared_ptr<TcpReaderWriter> tcpReaderWriter,
+      folly::Executor& executor)
+      : ExecutorBase(executor), tcpReaderWriter_(std::move(tcpReaderWriter)) {}
+
+  void onSubscribeImpl(
+      std::shared_ptr<Subscription> subscription) noexcept override {
+    if (tcpReaderWriter_) {
+      // no flow control at tcp level, since we can't know the size of messages
+      subscription->request(std::numeric_limits<size_t>::max());
```
Comment (alexmalyshev) [inline src/tcp/TcpDuplexConnection.cpp]: Is this necessary?  It seems like in the next line, `TcpReaderWriter::setOutputSubscription()` will also do `request(SIZE_T_MAX)`.

File: src/tcp/TcpDuplexConnection.cpp [hunk 2]
```diff
@@ -122,36 +116,113 @@ class TcpReaderWriter : public ::folly::AsyncTransportWrapper::WriteCallback,
...
+ public:
+  TcpInputSubscription(
+      std::shared_ptr<TcpReaderWriter> tcpReaderWriter,
+      folly::Executor& executor)
+      : ExecutorBase(executor), tcpReaderWriter_(std::move(tcpReaderWriter)) {
+    CHECK(tcpReaderWriter_);
+  }
+
+  void requestImpl(size_t n) noexcept override {
+    // TcpDuplexConnection doesnt support propper flow control
```
Comment (alexmalyshev) [inline src/tcp/TcpDuplexConnection.cpp]: Nit: `s|propper|proper|g`

File: src/tcp/TcpDuplexConnection.cpp (hunk 1 above)
Comment (lehecka) [inline src/tcp/TcpDuplexConnection.cpp]: good catch!
thank you

**Comment by lehecka** [comment]
replaced with https://github.com/rsocket/rsocket-cpp/pull/458

**Pull request closed without merging**